api/node/rust/interpreter/component_instance.rs

```diff
@@ -151,7 +156,15 @@ impl JsComponentInstance {
                             .unwrap();
 
                         if let Some(return_type) = &return_type {
-                            super::to_value(&env, result, return_type).unwrap()
+                            if let Ok(value) = super::to_value(&env, result, return_type) {
+                                return value;
+                            } else {
+                                eprintln!(
+                                    "Ndoe.js: callback {} throws an exception",
```

Comment:
Are you sure that the user provided callback threw an exception when to_value fails? Couldn't this be more likely related to the callback returning something that can't be converted to the expected type? It would be good to have a test here as well as an error message that gives a hint as to what failed.